app/src/main/java/com/eveningoutpost/dexdrip/Models/ReadingData.java

```diff
@@ -3,46 +3,47 @@
 // class from LibreAlarm
 
 import com.eveningoutpost.dexdrip.Models.UserError.Log;
+
+import com.eveningoutpost.dexdrip.utils.LibreTrendPoint;
 import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 
 public class ReadingData {
 
-    public PredictionData prediction;
-    public List<GlucoseData> trend;
-    public List<GlucoseData> history;
+    private static final String TAG = "ReadingData";
+    public List<GlucoseData> trend; // Per minute data.
+    public List<GlucoseData> history;  // Per 15 minutes data.
     public byte[] raw_data;
 
-    public ReadingData(PredictionData.Result result) {
-        this.prediction = new PredictionData();
-        this.prediction.realDate = System.currentTimeMillis();
-        this.prediction.errorCode = result;
-        this.trend = new ArrayList<>();
-        this.history = new ArrayList<>();
+    static final int ERROR_INFLUENCE = 4; //  The influence of each error
```

Comment:
Changed that, algoth I don't believe that extra space would cause any trouble.

Reply:
It would not cause any trouble but reduces memory consumption and it is a good practice to use the smallest possible data type in most cases, especially if it is a constant.